File: valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        char[] array1 = s.toCharArray();
        char[] array2 = t.toCharArray();
        Arrays.sort(array1);
        Arrays.sort(array2);
        s = new String(array1);
        t = new String(array2);

        if(s.length() != t.length()){
            return false;
        }

        for(int i = 0; i < s.length(); i++){
            
            if(s.charAt(i) != t.charAt(i)){
                return false;
            }
        }

        return true;
    }
}

File: 242-valid-anagram/valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    
    if(s.length !== t.length){
        return false;
    }

    const map = new Map();

    for(let i = 0; i < s.length; i++){

        if(!map.has(s.charAt(i))){
            map.set(s.charAt(i), 1);
        } else {
            map.set(s.charAt(i), map.get(s.charAt(i)) + 1);
        }
    }

    for(let j = 0; j < t.length; j++){
        if(!map.has(t.charAt(j))){
            return false
        } else {
            map.set(t.charAt(j), map.get(t.charAt(j)) -1 );
        }
    }

    if(Array.from(map.values()).every(value => value === 0)){
        return true;
    } else {
        return false;
    }

};